src/github.com/getlantern/netx/copy.go

```diff
@@ -0,0 +1,72 @@
+package netx
+
+import (
+	"io"
+	"net"
+	"sync/atomic"
+	"time"
+)
+
+var (
+	copyTimeout = 1 * time.Second
+)
+
+// BidiCopy copies between in and out in both directions, returning the errors
+// from copying to out and copying to in.
+func BidiCopy(out net.Conn, in net.Conn) (outErr error, inErr error) {
```

Comment:
Once this PR is complete, I'll use this same function on the proxy, which basically does the same thing as the client [here](https://github.com/getlantern/http-proxy/blob/master/httpconnect/httpconnect.go#L129).